### PR TITLE
Be case-insensitive when dealing with env. vars

### PR DESCRIPTION
**Pull request by palourde**
Signed-off-by: Simon Plourde <simon@sensu.io>

## What is this change?

Windows uses `Path` instead of `PATH` so be case-insensitive when looking for that env. variable.

## Why is this change necessary?

Closes https://github.com/sensu/sensu-go/issues/2057

## Does your change need a Changelog entry?

Added!

## Do you need clarification on anything?

Nope

## Were there any complications while making this change?

Nope

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Nope

### CODE REVIEW

**Comment by oferrigni** [comment]
Tests?

File: agent/assetmanager/manager.go [hunk 1]
```diff
@@ -77,15 +77,16 @@ func getSystemEnviron() []string {
 
 	for _, e := range env {
 		pair := strings.Split(e, "=")
-		key, _ := pair[0], pair[1]
+		// Transform the key to uppercase because Windows uses "Path" and not "PATH"
+		key := strings.ToUpper(pair[0])
 
-		if presentVars[key] != true {
+		if !presentVars[key] {
 			presentVars[key] = true
```
Comment (cyphus) [inline agent/assetmanager/manager.go]: Looks like we can unconditionally set presentVars[key] to true here.

**Comment by palourde** [comment]
@oferrigni We are calling `os.Environ()` within this function, which makes it very hard to test. I'll look into refactoring it so it's easier to test!

File: agent/assetmanager/manager.go (hunk 1 above)
Comment (palourde) [reply agent/assetmanager/manager.go]: Ha, good catch!

**Pull request merged**